### ISSUE TITLE
[10.0] Incorrect account move line in MO when using average cost price

### ISSUE THREAD

**Issue by CasVissers** [odoo/odoo#26217]
Impacted versions:
10.0

Steps to reproduce:
Product X - ( Real-time / Average cost price / Anglo saxon on)
Product X - Current stock qty = 1 / price =10
Product X - Create MO in which raw material total in price 5

We get two account moves:
1. Decrease the stock of the raw material + create booking for 5
2. Increase of the stock of product x with price 7,50 ((5+10)/2)

I.e. 

Current behavior:
The price used on the picking to increase stock of the item is the newly calculated standard_price. Which is incorrect.

Expected behavior:
The price used in the picking to increase stock of the item should containt the cost for specific MO (10)

Proposed fix:
stock_account - stock.py 
In product_price_update_before_done we already acknowlegde that behavior should be similar for production and purchase orders:
`for move in self.filtered(lambda move: move.location_id.usage in ('supplier', 'production') and move.product_id.cost_method == 'average'):`

But in:
```
            if self.product_id.cost_method == 'average':
                valuation_amount = cost if self.location_id.usage == 'supplier' and self.location_dest_id.usage == 'internal' else self.product_id.standard_price

```
We only check on supplier location, I tested the following change:
```
            if self.product_id.cost_method == 'average':
                valuation_amount = cost if self.location_id.usage in ('supplier','production') and self.location_dest_id.usage == 'internal' else self.product_id.standard_price
```

**Comment by pedrobaeza** [comment]
@CasVissers seeing current code, the fallback cost price is the current standard one, so I don't see the problem.

What you are expecting is that the cost price is computed from the accumulated cost of raw materials, but that's not performed, as it's not a correct formula, due to the rest of the manufacturing costs (workforce, machines, etc) is not taken into account, and it will be more confusing to have this half computation, so the cost price is kept in MO operations. I even doubt that these costs are correctly reflected on `cost` variable at that time. For what you want, you should develop custom extension assuming the limitations.

@jco-odoo can you confirm my guess?

**Comment by CasVissers** [comment]
@pedrobaeza 
Well in Anglo-Saxon accounting we will debit/credit our stock account input & output accounts while performing the MO. 

Similar to confirming an incoming shipment when we add stock to our stock account by first crediting the stock account input and afterwards debiting the stock account input when confirming the invoice. We need the values on input account to be the same. We actually use community modules to automatically reconcile them.

In an MO we are performing similar operations. On our production location we will set the specific stock input &output accounts to use (i.e. Production in progress). This so our stock moved in the production order will be either debited or credited to the stock account, credited in case of removing the item from stock (raw material), debited when adding an item to stock (finished goods or semi-finished goods). The main goal is to have our "production in progress" account clean/0 at the end of the MO. We can easily check this by comparing the amount of MO's in progress against the "production in progress" account.

You have a valid point about the workforce, machine costs etc. But since we don't make any booking for this while performing the MO, it doesn't make a difference in my approach. 

For sure that are more improvements which can be made (workforce etc), @krijn1981 maybe you have ideas for this? But this is a good start.

**Comment by CasVissers** [comment]
@nhomar 
How can something incorrectly handled by Odoo be marked as whishlist? The booking is incorrect as is. It will leave differences on input/output accounts which just can not be ignored. Accounting wise it's just incorrect.

**Comment by CasVissers** [comment]
And @pedrobaeza only noted some extra (possible) errors in the process. I just want a part of the process to be fixed so at least the bookings are right in more use cases. 
For Odoo's current process I see no use case.

**Comment by krijn1981** [comment]
I agree fully with Cas it is not wish, it is about filling the gaps in Odoo. 
That is what Odoo HQ is asking to us partners.

We are missing the base of accounting in this. Adding additional costs (like method direct/indirect costing and other methodes) is the next level/phase. If needed we can do a sparing session(s) for next coming Odoo versions, I'm open for that..

**Comment by CasVissers** [comment]
@krijn1981 
Seems like our opinion wasn't strong enough. Now merged in: #30766 

cc @pedrobaeza @nim-odoo
